paddleslim/nas/ofa/get_sub_model.py

```diff
@@ -68,40 +75,107 @@ def get_prune_params_config(graph, origin_model_config):
+                    else:
+                        in_chn = int(value.shape[0]) if in_exp == None else int(
+                            value.shape[0] * in_exp)
+                        out_chn = int(value.shape[
+                            1]) if out_exp == None else int(value.shape[1] *
+                                                            out_exp)
+                        prune_value = super_value[:in_chn, :out_chn, ...] \
+                                         if super_model_sd != None else value[:in_chn, :out_chn, ...]
+                else:
+                    out_chn = int(value.shape[0]) if param_config[param.name][
+                        0] == None else int(value.shape[0] *
+                                            param_config[param.name][0])
+                    prune_value = super_value[:out_chn, ...] \
+                                     if super_model_sd != None else value[:out_chn, ...]
+
+            else:
+                prune_value = super_value if super_model_sd != None else value
+
+            p = t_value._place()
+            if p.is_cpu_place():
+                place = core.CPUPlace()
+            elif p.is_cuda_pinned_place():
+                place = core.CUDAPinnedPlace()
             else:
-                out_chn = int(value.shape[0]) if param_config[param.name][
-                    0] == None else int(value.shape[0] *
-                                        param_config[param.name][0])
-                prune_value = super_value[:out_chn, ...] \
-                                 if super_model_sd != None else value[:out_chn, ...]
-        else:
-            prune_value = super_value if super_model_sd != None else value
-
-        p = t_value._place()
-        if p.is_cpu_place():
-            place = core.CPUPlace()
-        elif p.is_cuda_pinned_place():
-            place = core.CUDAPinnedPlace()
-        else:
-            place = core.CUDAPlace(p.gpu_device_id())
-        t_value.set(prune_value, place)
-        if param.trainable:
-            param.clear_gradient()
+                place = core.CUDAPlace(p.gpu_device_id())
+            t_value.set(prune_value, place)
+            if param.trainable:
+                param.clear_gradient()
+
+
+def _find_weight_ops(op, graph, weights):
+    pre_ops = graph.pre_ops(op)
+    for pre_op in pre_ops:
+        if pre_op.type() in WEIGHT_OP:
+            for inp in pre_op.all_inputs():
+                if inp._var.persistable:
+                    weights.append(inp._var.name)
+            return weights
+        return _find_weight_ops(pre_op, graph, weights)
+
+
+def _find_pre_elementwise_add(op, graph):
+    same_ss_per_op = []
+    pre_ops = graph.pre_ops(op)
+    for pre_op in pre_ops:
+        if pre_op.type() in WEIGHT_OP:
+            return
+        same_ss_per_op = _find_weight_ops(pre_op, graph, same_ss_per_op)
+    return same_ss_per_op
+
+
+def check_ss(graph):
```

Comment:
ss -> search_sapce,   need more clear name  and add comments

Reply:
done, thanks